src/utils/pqUtils.ts

```diff
@@ -110,10 +111,56 @@ const validateQueryName = (queryName: string): void => {
         throw new Error(EmptyQueryNameErr);
     }
 };
+
+const validateMultipleQueryNames = (queries: ConnectionOnlyQueryInfo[],  loadedQueryName: string): string[] => {
+    const queryNames: string[] = [];
+    const cleanedLoadedQueryName: string = loadedQueryName.trim().toLowerCase();
+    queries.forEach((query: ConnectionOnlyQueryInfo) => {
+        if (query.queryName) {
+            validateQueryName(query.queryName);
+            const cleanedQueryName: string | undefined = query.queryName.trim().toLowerCase();
+            if (queryNames.includes(cleanedQueryName) || cleanedQueryName === cleanedLoadedQueryName) {
+                throw new Error(queryNameAlreadyExistsErr);
+            }
+
+            queryNames.push(cleanedQueryName);
+        } 
+    });
+    
+    return queryNames;
+};
+
+const assignQueryNames = (queries: ConnectionOnlyQueryInfo[], loadedQueryName: string, queryNames: string[]):  ConnectionOnlyQueryInfo[] => {
+    // Generate unique name for queries without a name
+    queries.forEach((query: ConnectionOnlyQueryInfo) => {
+        if (!query.queryName) {
+            query.queryName = generateUniqueQueryName(queryNames, loadedQueryName);
```

Comment:
Do we want to generate meaningful query name? for example to the connection only queries -> connection only query-1 (just an example)